docs/guides/new-policies.md

```diff
@@ -11,3 +11,79 @@ The output of the tool will tell you where the important files are!
 ## Add plugin name to the configuration
 
 Enabled plugin configuration is in `pkg/plugins/policies/policies.go`. Plugins name is equals to `KumactlArg` in file `zz_generated.resource.go`. It's important to place the plugin in the correct place because the order of executions is important.
+
+## How to map API to a Go struct
+
+Field is _mergeable_ if it is directly in `default` struct or any sub-struct.
```

Comment:
Is "mergeable" defined anywhere?

Reply:
Only in this memo. I don't think it makes sense to expose it in docs, it's useful only when you create a new policy

Reply:
But what does it mean that something is "mergeable"? I'm not quite sure. I just see:

* Mergeable: a, b, c, d
* Not mergeable: e, f, g, h

but what consequence does this have for those fields?

Reply:
It just means mergeable fields participate in [configuration merging](https://kuma.io/docs/2.0.x/policies/targetref/#merging-configuration)

Reply:
Right ok, that could be linked here. To be honest I'm not sure there's anything special about lists here, it's ultimately kind of obfuscating the nature of merging. A list is just a type that isn't a struct, so of course it's not mergeable?

Reply:
But also it's an internal doc so probably not worth discussing too much :grinning:

Reply:
> A list is just a type that isn't a struct, so of course it's not mergeable

The list itself is mergeable, but elements inside the list are not mergeable. 

But you're right nothing special about lists, I just wanted to show clearly that if the field is mergeable then it's always optional.

Reply:
I think the explanation with lists is unnecessary and confusing and it should just show the examples of making a go field optional. And what's the difference between the 3 types of examples? Aren't they all just with and without pointer?

Reply:
Isn't the message really just make sure you wrap your the type of your field with a pointer if that field belongs to a mergeable struct?

Reply:
>And what's the difference between the 3 types of examples?

It might be not obvious that you need to add a pointer to the slice. That's why it makes sense to show the example for lists. But I agree that Struct and Basic types could be merged into 1 section.

> Aren't they all just with and without pointer?

Also `omitempty`.

> Isn't the message really just make sure you wrap your the type of your field with a pointer if that field belongs to a mergeable struct?

Even if the field doesn't belong to mergeable struct it still can be either `required` or `optional`.

Reply:
> > Aren't they all just with and without pointer?
> 
> Also omitempty.
> 
> > Isn't the message really just make sure you wrap your the type of your field with a pointer if that field belongs to a mergeable struct?
> 
> Even if the field doesn't belong to mergeable struct it still can be either required or optional.

Right, sure, I just mean that the whole thing is basically a very simple "fields in a mergeable struct must be optional. Make the fields optional the same way you always make a field optional using JSON and go". Like the whole thing makes it seem way more complicated than it is.

> It might be not obvious that you need to add a pointer to the slice. 

Maybe but that goes for any time you want an optional list. `[] != nil`

Reply:
What's complicated about this? There are just 6 examples, I can reduce it to 4:
* List
  * Required
  * Optional
* Struct/Basic type
  * Required
  * Optional

Isn't it easier to look at examples than trying to figure out "the same way you always make a field optional using JSON and go"?

Reply:
It isn't complicated at all, this doc just makes it _seem_ complicated. It's noisy. The discussion on mergeable fields is not helpful. It's confusing because when I read this it leads me to believe there's more happening than just "add a pointer and omitempty, like always." Pretending it's something other than that is confusing, "do it like idiomatic go" tells me everything I need to know. What about the 3 sentences:

> See the docs[kuma.io link] for a discussion on what it means for structs to be merged and when they're merged. When writing a `targetRef` policy, fields in a mergeable struct must be optional. Make the fields optional by wrapping with a pointer and including `omitempty`.

I feel like it's very rare that someone hasn't seen a go struct with a JSON-optional field using a pointer wrapper and `omitempty` and is writing a new policy, it's all over the current resources and k8s resources and otherwise everywhere but an example wouldn't hurt I suppose.

You can merge it, I really don't mind, it's an internal doc so I won't say it shouldn't be merged, I'm just trying to make it more helpful to anyone reading it.

Reply:
> "do it like idiomatic go" tells me everything I need to know

Is there an idiomatic Go way of mapping json to go-struct so we could reference it here? TBH I didn't find it. 

If you set `omitempty` without the pointer it's also "optional" but it won't work with merging. Also, you can make optional arrays differently using controller-gen annotation (that's actually how it's done today in most new policies):
```go
// +optional
// +nullable
MyArrayField []ItemType `json:"myArrayField"`
```
but we want to emphasize in this doc to use slice pointer:
```go
MyArrayField *[]ItemType `json:"myArrayField,omitempty"`
```